src/partition/repart.c

```diff
@@ -5275,6 +5336,26 @@ static int do_make_directories(Partition *p, const char *root) {
         return 0;
 }
 
+static int do_make_symlinks(Partition *p, const char *root) {
+        int r;
+
+        assert(p);
+        assert(root);
+
+        STRV_FOREACH_PAIR(path, target, p->make_symlinks) {
+                _cleanup_free_ char *resolved = NULL;
+
+                r = chase(*path, root, CHASE_PREFIX_ROOT|CHASE_NONEXISTENT, &resolved, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to resolve %s in %s", *path, root);
+
+                if (symlink(*target, resolved) < 0)
+                        return log_error_errno(errno, "Failed to create symlink at %s to %s: %m", resolved, *target);
```

Comment:
would be nicer to use CHASE_PARENT get an O_PATH fd, and then path_extract_filename() + symlinkat().